lib/InstModel/GenericInstModel/GenericInstModel.cpp

```diff
@@ -159,19 +168,33 @@ std::vector<Token *> GenericInstModel::extractToken(llvm::CallInst *CI) {
   }
   return TokenVec;
 }
+/// extractPHINodeToken - Returns the alias objects for PHInode instruction
+std::vector<Token *> GenericInstModel::extractPHINodeToken(llvm::Instruction *Inst){
+	std::vector<Token *> TokenVec;
+	TokenVec.push_back(this->getTokenWrapper()->getToken(Inst));
+	if(Inst->getOperand(0)->hasName()){
```

Comment:
Do not use hasName; I think you are using hasName to check if it a variable or a constant? if yes, prefer checking if it is ConstantInt, like we did in the StoreInst.

Reply:
I have done this with the help of llvm::Constant

Reply:
Yes, that will be great. Though, hasName will also work fine but it will add an explicit dependency to run instnamer or something instnamer like to get it working.